app/views/users/_form.html.haml

```diff
@@ -0,0 +1,6 @@
+= form_for @current_user do |f|
+  = f.label 'user[timezone_id]', I18n.t('users.form.label'), class: 'form-label'
```

Comment:
a couple of handy tips to save you some typing in future: 

1. Rails has a built-in helper method `t()`, so you don't need the `I18n.` in views
2. Rails will also default the translation scope to the path of the current view template 

So in this case your `I18n.t('users.form.label')` could just be `t('.label')`